lux/core/frame.py

```diff
@@ -101,6 +101,12 @@ def f(*args, **kwargs):
     def history(self):
         return self._history
 
+    @property
+    def data_type(self):
+        if not self._data_type:
+            self.maintain_metadata
```

Comment:
This line needs to be actually called right?
`self.maintain_metadata()`

Reply:
Oooooops -- fixed in the next push